app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -9,16 +9,10 @@ def create_resource(type, _id = nil, data = {})
     end
 
     def edit_resource(type, id, data = {})
-      raise BadRequestError, "Must specify an id for editing a #{type} resource" unless id
-
-      cloud_volume = resource_search(id, type)
-
-      raise BadRequestError, cloud_volume.unsupported_reason(:update) unless cloud_volume.supports?(:update)
-
-      task_id = cloud_volume.update_volume_queue(User.current_user, data)
-      action_result(true, "Updating #{cloud_volume.name}", :task_id => task_id)
-    rescue => err
-      action_result(false, err.to_s)
+      api_resource(type, id, "Updating") do |cloud_volume|
+        ensure_supports(type, cloud_volume, :update)
```

Comment:
Does it make sense to push down the supports check?  I'm pictureing something like:

```suggestion
      api_resource(type, id, "Updating", :supports => :update) do |cloud_volume|
```

Reply:
Ok, so there is a special case when the action is not the same as the supports action name. I didn't know how to handle this but turns out this is a rare case anyway.
```
      enqueue_ems_action(type, id, "Scanning", :method_name => "scan", :supports => :smartstate_analysis, :role => "smartstate")
```

In our case, I think we want to use `:supports => type` for most (if not all) of these cases.

I can put together a PR with the supports use case.

Reply:
@Fryguy Created #1135 - Is this what you had in mind?